gwsumm/data/range.py

```diff
@@ -117,21 +117,35 @@ def get_range_spectrogram(channel, segments, config=None, cache=None,
             add_spectrogram(outspec if 'energy' in rangekwargs else
                             outspec**(1/2.), key=key)
 
-    if return_:
-        return globalv.SPECTROGRAMS.get(key, SpectrogramList())
+    if not return_:
+        return
+
+    # return correct data, according to state segment
+    out = SpectrogramList()
+    for specgram in globalv.SPECTROGRAMS[key]:
+        for seg in segments:
+            if abs(seg) < specgram.dt.value:
+                continue
+            if specgram.span.intersects(seg):
+                common = specgram.span & type(seg)(seg[0],
+                                                   seg[1] + specgram.dt.value)
+                s = specgram.crop(*common)
+                if s.shape[0]:
+                    out.append(s)
+    return out.coalesce()
 
 
 @use_segmentlist
 def get_range_spectrum(channel, segments, config=None, cache=None, query=True,
                        nds=None, return_=True, nproc=1, datafind_error='raise',
                        frametype=None, stride=60, fftlength=None, overlap=None,
-                       method=None, which='all', **rangekwargs):
+                       method=None, which='all', state='', **rangekwargs):
```

Comment:
I prefer not having an empty string be the "no state given" value. I suggest using `None` and then I'll suggest another change on L146 assuming you're fine this suggestion
```suggestion
                       method=None, which='all', state=None, **rangekwargs):
```